src/components/Reportees/index.jsx

```diff
@@ -6,58 +6,159 @@ import TableBody from '@material-ui/core/TableBody';
-                        href={countLink.link}
-                        target="_blank"
-                        rel="noopener noreferrer"
-                      >
-                        {countLink.count}
-                      </a>
-                    )}
-                  </TableCell>
-                );
-              })}
-            </TableRow>
-          ))}
-      </TableBody>
-    </Table>
-  </div>
-);
 
+function desc(a, b, orderBy) {
+  const valueA = a[orderBy];
+  const valueB = b[orderBy];
+
+  const countA = valueA ? valueA.count : 0;
+  const countB = valueB ? valueB.count : 0;
+
+  if (countB < countA) {
+    return -1;
+  }
+
+  if (countB > countA) {
+    return 1;
+  }
+
+  return 0;
+}
+
+function stableSort(array, cmp) {
+  const stabilizedThis = array.map((el, index) => [el, index]);
+  stabilizedThis.sort((a, b) => {
+    const order = cmp(a[0], b[0]);
+    if (order !== 0) return order;
+    return a[1] - b[1];
+  });
+  const result = stabilizedThis.map(el => el[0]);
+  return result;
+}
+
+function getSorting(order, orderBy) {
+  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy);
+}
+
+
+class Reportees extends React.PureComponent {
+  state = {
+    orderBy: 'cn',
+    order: 'asc',
+  };
+
+  getMergedProps() {
+    const { metrics, partialOrg, ldapEmail } = this.props;
+
+    // filter out the manager
+    const reportees = Object.values(partialOrg)
+      .filter(({ mail }) => mail !== ldapEmail);
+
+    // add metrics
+    const reporteesWithMetrics = reportees.map(reportee => ({
+      ...reportee,
+      ...metrics[reportee.bugzillaEmail],
+    }));
+
+    return reporteesWithMetrics;
+  }
+
+  createSortHandler = property => (event) => {
+    this.handleRequestSort(event, property);
+  };
+
+  handleRequestSort = (event, property) => {
+    const { order, orderBy } = this.state;
+
+    let newOrder = 'desc';
+    if (orderBy === property && order === 'desc') {
+      newOrder = 'asc';
+    }
+
+    this.setState({
+      orderBy: property,
+      order: newOrder,
+    });
+  };
+
+  render() {
+    const { classes } = this.props;
+
+    const { order, orderBy } = this.state;
+
+    return (
+      <div className={classes.root}>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell />
+              {Object.entries(CONFIG.reporteesMetrics).map(([metricUid, { label }]) => (
+                <TableCell
+                  key={metricUid}
+                  align="right"
+                  sortDirection={orderBy === metricUid ? order : false}
+                >
+                  <Tooltip
+                    title="Sort"
+                    placement="bottom-end"
+                  >
+                    <TableSortLabel
+                      active={orderBy === metricUid}
+                      direction={order}
+                      onClick={
+                          this.createSortHandler(metricUid)
```

Comment:
Instead of having to pass in `metricUid`, you could just set an `id` to the component:

```jsx
<TableSortLabel
  id={metricUid}
  onClick={this.handleHeaderClick}
  // ...
/>


handleHeaderClick = ({ target }) => {
    const sortBy = target.id;
    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';

    this.setState({ sortBy, sortDirection });
};
```

Reply:
We could, but this means storing some state in the DOM, which I don't like so much. Also I think this is more difficult to handle over time, and generally this pattern works only well with strings (which we have here).
So in this case this is possible, so if you really want it we could do it, but I'd like to understand the upsides of the change :)

Reply:
Something like `onClick={this.handleHeaderClick(metricUid)}` would initiate a function call for each `TableSortLabel` on page load and on every page render. Embedding `metricUid` in the `id` attribute would remove these unnecessary calls.

It's not a big deal if it's kept like it is :)

Reply:
There are an awful lot of other function calls (all JSX calls to start with), so I think this one is pretty much harmless and nearly free :-)

Reply:
TBH in a more important project I'd probably do it differently and either memoize the result of the handlers (because after all they're always the same for each update), but here I'm a bit lazy...